stan/math/rev/fun/unit_vector_constrain.hpp

```diff
@@ -34,7 +34,7 @@ inline auto unit_vector_constrain(const T& y) {
   arena_t<T> arena_y = y;
   arena_t<promote_scalar_t<double, T>> arena_y_val = arena_y.val();
 
-  const double r = arena_y_val.norm();
+  const double r = norm2(arena_y_val);
```

Comment:
For `double` arguments, it's probably better to use the built-in Eigen functions as they're more highly optimized for this.  Of course, `norm2` should delegate to this `.norm()` for values in the vector case.